_posts/2023-06-04-not-generating-constructors.md

```diff
@@ -0,0 +1,411 @@
+bounds in the C++ standard.
+
+Much has been written about the memory unsafety of lack of bounds checking, and we (as in the
+software industry) have known ways to solve this simple issue for decades now.
+* Here's an article from 1996 (almost 30 years ago)
+[proposing a bounds-checked structure](https://ecs.syr.edu/faculty/fawcett/Handouts/cse775/presentations/BruceMcKinneyPapers/safeArrays.htm)
+for interacting with
+[OLE](https://learn.microsoft.com/en-us/cpp/mfc/ole-background).
+* Here's a [NIST paper titled "C++ in Safety Critical Systems"](
+https://upload.wikimedia.org/wikipedia/commons/5/50/C%2B%2B_in_safety_critical_systems_%28IA_cinsafetycritica5769bink%29.pdf)
+from 1995 which states "Class libraries that provide smart (safe) pointers and array bounds
+checking should be used."
+
+In 2022, The Clang/Libc++ project accepted the
+[C++ Buffer Hardening](https://discourse.llvm.org/t/rfc-c-buffer-hardening/65734) proposal that
+makes it possible to enable hardening checks inside these standard library types with bounds.
+Though it requires you to build your own standard library. The one that shipped with your operating
+system likely [doesn't have this enabled](https://godbolt.org/z/Y4n99dx5j). At least your
+[web browser](https://bugs.chromium.org/p/chromium/issues/detail?id=1335422) may have this enabled
+(Chrome has since Q1 2023, but I don't know about any other Browser,
+[Chromium-derived or not](https://chromium.googlesource.com/chromium/src/+/master/docs/security/faq.md#how-can-i-know-which-fixes-to-include-in-my-downstream-project)).
+
+The same proposal introduces the [-Wunsafe-buffer-usage](
+https://clang.llvm.org/docs/DiagnosticsReference.html#wunsafe-buffer-usage)
+warning in Clang which you can enable to ban pointer arithmetic and
+which has the intention of converting all those native pointers into view types, which pair a size
+with the pointer through the type system. See, the problem is that even if you enable bounds
+checking in `std::span` and `std::string_view`, the majority of C++ code really doesn't use those
+types, our industry's code bases are still full of pointers to arrays. So this warning gives a
+mechanism to stard migrating away from pointers, _and then prevent backsliding_.
+
+## Performance implications
+
+Introducing memory safety into C++ codebases always comes with some performance degredation, because
+there's little you can check at compile time. So it means checking for bad program states when the
+programming is running. But these performance changes are not the kind you as a user will see or
+feel. But the kind that shows up on benchmarks and that software teams then have great a many
+[discussions about](https://groups.google.com/a/chromium.org/g/platform-architecture-dev/c/hdMBHEYZLMA).
+And these often end up blocking meaningful improvements to the safety of users and their data (for
+better or for worse).

```

Comment:
What's the better side?

Reply:
heh heh. I was not sure about injecting my moral stance here or not and kinda want people to make up their mind a little but idk?

Reply:
I think it's actually just stronger to drop the FBOFW.